LogicalTypes.md

```diff
@@ -609,6 +609,17 @@ that is neither contained by a `LIST`- or `MAP`-annotated group nor annotated
 by `LIST` or `MAP` should be interpreted as a required list of required
 elements where the element type is the type of the field.
 
+```
```

Comment:
I would suggest putting these examples, and the paragraph above, in a separate subsection, for example:
````markdown
Implementations should use either `LIST` and `MAP` annotations _or_ unannotated
repeated fields, but not both. When using the annotations, no unannotated
repeated types are allowed.

### Unannotated repeated fields

A repeated field that is neither contained by a `LIST`- or `MAP`-annotated
group nor annotated by `LIST` or `MAP` should be interpreted as a
required list of required elements where the element type is the type of
the field.

```
// List<Integer> (non-null list, non-null elements)
repeated int32 num;

// List<Tuple<Integer, String>> (non-null list, non-null elements)
repeated group my_list {
  required int32 num;
  optional binary str (STRING);
}
```

It is heavily recommended that writers use either the LIST or MAP annotations,
as outlined below, to make the intent clearer to readers.
````

Reply:
I agree that it's slightly odd, but we would otherwise need a separate doc for interpretation of schemas as you say (and move any relevant contents out of `LogicalTypes.md`).

Reply:
It looks strange to make it a subsection if the unannotated repeated fields are not deprecated. If we deprecate it, `Backward-compatibility rules` section can be organized as below:
- 3-level structure with different field names
- 2-level structure
- Unannotated repeated fields

It would be better to deprecate it in another change. Therefore I tend to keep it as is.

Reply:
> I'm not sure either why `parquet-rs` chosen to write "1-level" lists.

It doesn't by default. It takes user intervention to pass a schema with 1-level lists to the record writer. The table writer will use the proper 3-level structure AFAICT.

Reply:
Thanks, @etseidl.

Related to protobuf (the parquet-java binding for protobuf) this is the other way around currently. There is a flag for converting repeated protobuf elements to "3-level" lists but it is false by default, so it converts directly to similar repeated elements in the parquet schema ("1-level" list).
Anyways it seems that neither parquet-rs nor parquet-java (protobuf) should have an issue creating proper "3-level" lists.